app/controllers/concerns/spree/admin/translatable.rb

```diff
@@ -16,12 +16,11 @@ def save_translation_values
         translation_params = params[:translation]
 
         current_store.supported_locales_list.each do |locale|
-          I18n.with_locale(locale) do
-            translation_params.each do |attribute, translations|
-              @object.public_send("#{attribute}=", translations[locale])
-            end
-            @object.save!
+          translation = @object.translations.find_or_create_by(locale: locale)
```

Comment:
Maybe `find_or_initialize_by` would be more fitting in this context. It calls `new` instead of `create` so there's no DB write.
```suggestion
          translation = @object.translations.find_or_initialize_by(locale: locale)
```